pkg/scalers/honeycomb_scaler.go

```diff
@@ -0,0 +1,300 @@
+package scalers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/go-logr/logr"
+	v2 "k8s.io/api/autoscaling/v2"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+
+	"github.com/kedacore/keda/v2/pkg/scalers/scalersconfig"
+	kedautil "github.com/kedacore/keda/v2/pkg/util"
+)
+
+const (
+	honeycombScalerName        = "honeycomb"
+	honeycombBaseURL           = "https://api.honeycomb.io/1"
+	maxPollAttempts            = 5
+	initialPollDelay           = 2 * time.Second
+	honeycombQueryResultsLimit = 10000
+)
+
+type honeycombScaler struct {
+	metricType v2.MetricTargetType
+	metadata   honeycombMetadata
+	httpClient *http.Client
+	logger     logr.Logger
+}
+
+type honeycombMetadata struct {
+	APIKey              string                 `keda:"name=apiKey, order=authParams;triggerMetadata"`
```

Comment:
We don't support reading secrets from metadata
```suggestion
	APIKey              string                 `keda:"name=apiKey, order=authParams"`
```